cli/src/cmd_disk.rs

```diff
@@ -93,461 +78,107 @@ pub struct CmdDiskImport {
-            let result = handle.await?;
-            results.push(result);
         }
+    });
 
-        if results.iter().any(|x| x.is_err()) {
-            // If any of the upload threads returned an error, unwind the disk.
-            eprintln_nopipe!("one of the upload threads failed");
-            self.unwind_disk_bulk_write_stop(client).await?;
-            self.unwind_disk_finalize(client).await?;
-            self.unwind_disk_delete(client).await?;
-            bail!("one of the upload threads failed");
-        }
-
-        // wait for upload threads to complete, then finish the progress bar
-        pb.finish();
-
-        // Stop the bulk write process
-        let stop_bulk_write_response = client
-            .disk_bulk_write_import_stop()
-            .project(&self.project)
-            .disk(self.disk.clone())
-            .send()
-            .await;
-
-        if let Err(e) = stop_bulk_write_response {
-            eprintln_nopipe!("stopping the bulk write process failed with {:?}", e);
-
-            // Attempt to unwind the disk, although it will probably fail - the
-            // first step is to stop the bulk write process!
-            self.unwind_disk_bulk_write_stop(client).await?;
-            self.unwind_disk_finalize(client).await?;
-            self.unwind_disk_delete(client).await?;
-
-            return Err(e.into());
-        }
+    Ok(pb)
+}
 
-        // Finalize the disk, optionally making a snapshot
-        let request = client
-            .disk_finalize_import()
-            .project(&self.project)
-            .disk(self.disk.clone())
-            .body(FinalizeDisk {
-                snapshot_name: self.snapshot.clone(),
-            });
+fn watch_for_ctrl_c(handle: DiskImportHandle, pb: ProgressBar) {
+    tokio::spawn(async move {
+        let mut force_shutdown = false;
+        let mut handle = Some(handle);
 
-        let finalize_response = request.send().await;
+        loop {
+            signal::ctrl_c().await.expect("Failed to listen for CTRL+C");
 
-        if let Err(e) = finalize_response {
-            eprintln_nopipe!("finalizing the disk failed with {:?}", e);
+            if force_shutdown {
+                eprintln_nopipe!("Shutting down immediately.\nSee https://docs.oxide.computer/guides/troubleshooting#_cant_delete_disk_after_canceled_image_import for instructions on removing the disk.");
 
-            // Attempt to unwind the disk, although it will probably fail - the
-            // first step is to finalize the disk!
-            self.unwind_disk_finalize(client).await?;
-            self.unwind_disk_delete(client).await?;
+                // Match exit code used by Bash on Unix, 128 + 2 (SIGINT).
+                process::exit(130);
+            }
+            force_shutdown = true;
 
-            return Err(e.into());
-        }
+            pb.finish_and_clear();
+            pb.println("Cleaning up disk. Press CTRL+C again to exit immediately.");
```

Comment:
I would suggest we omit the "... exit immediately" text in that basically we don't want people to do that... and in my experience they don't need an invitation to ^C

Reply:
Ha, very true